src/Tribe/Data_API.php

```diff
@@ -0,0 +1,517 @@
+			if ( ! isset( $services['class'] ) && is_numeric( $post_id ) ) {
+				return Tribe__Tickets__Tickets::get_event_attendees( $post_id );
+			} elseif ( ! isset( $services['class'] ) && ! is_numeric( $post_id ) ) {
+				return array();
+			}
+
+			return $services['class']::get_instance()->get_attendees_by_id( $post_id, $services['post_type'] );
+
+		}
+
+		/**
+		 * Check if attendee(s) have meta data
+		 *
+		 * @param $has_meta
+		 * @param $attendees
+		 * @param $has_meta_fields
+		 *
+		 * @return bool
+		 */
+		protected function attendees_meta_check( $has_meta, $attendees ) {
+
+			foreach ( $attendees as $attendee ) {
+				if ( isset( $attendee['attendee_meta'] ) && ! empty( $attendee['attendee_meta'] ) ) {
+					$has_meta = true;
+				}
+			}
+
+			return $has_meta;
+
+		}
+
+
+		/**
+		 * Check if a order key passed exists and return attendee object name
+		 *
+		 * @param $order_key
+		 *
+		 * @return string
+		 */
+		protected function check_rsvp_order_key_exists( $order_key ) {
+
+			$attendees_query = $this->query_by_rsvp_order_key( $order_key );
+			if ( ! $attendees_query->have_posts() ) {
+				return '';
+			}
+
+			return Tribe__Tickets__RSVP::ATTENDEE_OBJECT;
+
+		}
+
+		/**
+		 * Get the rsvp order key from a post id
+		 *
+		 * @param $post_id
+		 *
+		 * @return mixed
+		 */
+		protected function get_rsvp_order_key( $post_id ) {
+
+			$order_key = get_post_meta( $post_id, Tribe__Tickets__RSVP::get_instance()->order_key, true );
+			if ( ! $order_key ) {
+				return $post_id;
+			}
+
+			return $order_key;
+
+		}
+
+		/**
+		 * Get a post id when passing a rsvp order key
+		 * Since all rsvp orders will be from one post,
+		 * we only need to return the first match
+		 *
+		 * @param $order_key
+		 *
+		 * @return false|int|string
+		 */
+		protected function get_rsvp_post_id_from_order_key( $order_key ) {
+
+			$attendees_query = $this->query_by_rsvp_order_key( $order_key, 1 );
+			if ( ! $attendees_query->have_posts() ) {
+				return '';
+			}
+
+			$post_id = '';
+
+			while ( $attendees_query->have_posts() ) {
+				$attendees_query->the_post();
+				$post_id = get_the_ID();
+			}
+
+			wp_reset_postdata();
+
+			return $post_id;
+
+		}
+
+
+		/**
+		 * Query RSVP Orders by the Order Key
+		 *
+		 * @param        $order_key
+		 * @param string $post_per_page
+		 *
+		 * @return WP_Query
+		 */
+		protected function query_by_rsvp_order_key( $order_key, $post_per_page = '-1' ) {
+
+			$attendees_query = new WP_Query( array(
+				'posts_per_page' => $post_per_page,
+				'post_type'      => Tribe__Tickets__RSVP::ATTENDEE_OBJECT,
+				'meta_key'       => Tribe__Tickets__RSVP::get_instance()->order_key,
+				'meta_value'     => esc_attr( $order_key ),
+				'orderby'        => 'ID',
+				'order'          => 'ASC',
+			) );
+
+			return $attendees_query;
+
+		}
+
+
+		/********** SINGLETON FUNCTIONS **********/
+
+		/**
+		 * Instance of this class for use as singleton
+		 */
+		private static $instance;
```

Comment:
Rather than add singleton code in here, why not use `tribe_singleton()` and related helpers to manage this (especially as the constructor remains public)?